internal/metrics/metrics.go

```diff
@@ -0,0 +1,174 @@
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+)
+
+const (
+	CompactorEnabled  string = "compactor_enabled"
+	CompactorDisabled string = "compactor_disabled"
+	TenantLimits      string = "tenant_limits"
+	GlobalLimits      string = "global_limits"
+)
+
+var (
+	sizeTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_cluster_sizes",
+			Help: "Number of clusters that are using a specific size configuration",
+		},
+		[]string{"size"},
+	)
+
+	globalStreamTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_max_global_streams",
+			Help: "Total of max global streams defined by the custom resource",
```

Comment:
Consider [best practices](https://prometheus.io/docs/practices/naming/) on metric and label naming, e.g. if the description refers to a total then the metric name should probably look like `lokistack_max_global_streams_total`.

Reply:
See https://www.robustperception.io/on-the-naming-of-things